frontend/src/components/doctor/NewMedicalRecordForm.jsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import PatientCard from "./PatientCard";
+import { Input, Textarea, Button, addToast } from "@heroui/react";
+import { useState } from "react";
+import { createMedicalRecord } from "../../api/medicalRecordFunctions";
+import { useNavigate } from "react-router-dom";
+
+export default function NewMedicalRecordForm({
+  patientBody,
+  newMedicalRecord,
+}) {
+  //Form fields
+  const [notes, setNotes] = useState();
+  const [diagnosis, setDiagnosis] = useState();
+
+  const navigate = useNavigate();
+
+  async function createNewMedicalRecord() {
+    if (!diagnosis) {
+      addToast({
+        title: "There was an error creating the medical record",
+        description:
+          "Please add a diagnosis before creating the medical record",
+        color: "danger",
+        timeout: 10000,
+      });
+
+      return;
+    }
+
+    try {
+      const newRecord = await createMedicalRecord({
+        ...newMedicalRecord,
+        diagnosis,
+        notes,
+      });
+    } catch (error) {
+      console.error(error);
+      addToast({
+        title: "There was an error creating the medical record",
+        description: "Please try again",
+        color: "danger",
+        timeout: 10000,
+      });
+    } finally {
+      addToast({
+        title: "The medical record was added successfully",
+        color: "success",
+        timeout: 10000,
```

Comment:
Similar to the earlier comment, let's parameterize this timeout as they're mostly the same